### DISCUSSION TITLE
Is there a way to autodetect if pager supports styles/colors?

### DISCUSSION THREAD

**Discussion by yashrathi-git** [Textualize/rich#1475] [Q&A] [answered]
Hi,
Some terminal emulators don't support rich emoji and pager style. `rich.pager` don't check if pager support styles or not. Is there any way to autodetect support for characters like emoji and pager styles?

**Comment by willmcgugan** [comment, marked as answer]
Not that I know of. It could be done via an env var.

**Reply by yashrathi-git** [reply]
I found out that [click](https://github.com/pallets/click/blob/74d1b081f6e666413ee81d68b26d48a793e0a060/src/click/termui.py#L273) supports auto-detection, the function [click.echo_via_pager is here](https://github.com/pallets/click/blob/74d1b081f6e666413ee81d68b26d48a793e0a060/src/click/termui.py#L273). I am unable to understand what they are doing. This is what I did:
```py
from contextlib import contextmanager
from rich.console import Console
import click
from typing import Optional

console=Console()


@contextmanager
def rich_click_pager(color: Optional[bool] = None):
    with console.capture() as capture:
        yield
    printed_text = capture.get()
    click.echo_via_pager(printed_text, color=color)

with rich_click_pager():
    console.print('[red on white]test[/]'* 1000)
```

If the color is None click autodetect the support for color.  I haven't been able to test it as I don't have a windows machine.

**Reply by willmcgugan** [reply]
Looks like it tries make a best guess by enumerating all the possible pagers that may be in use. Which will probably be correct the vast majority of the time, but may not work if you have a pager the click authors didn't test for. Still, perhaps Rich could adopt that.

Your solution is good. You could also define a [Pager](https://github.com/willmcgugan/rich/blob/master/rich/pager.py) class that uses click.

**Reply by yashrathi-git** [reply]
alright, thank you :)
